packages/astro/src/core/errors/errors-data.ts

```diff
@@ -532,6 +532,26 @@ See https://docs.astro.build/en/guides/server-side-rendering/ for more informati
 			)} are supported for optimization.`,
 		hint: "If you do not need optimization, using an `img` tag directly instead of the `Image` component might be what you're looking for.",
 	},
+	/**
+	 * @docs
+	 * @see
+	 * - [`getStaticPaths()`](https://docs.astro.build/en/reference/api-reference/#getstaticpaths)
+	 * - [`params`](https://docs.astro.build/en/reference/api-reference/#params)
+	 * @description
+	 * You should add an additional extension to the endpoint's filename 
+	 * and A dynamic endpoint with `getStaticPaths` should not return `params` with `undefined`.
```

Comment:
Thanks for updating this part @wulinsheng123. I'm checking the errors-data readme again, I think it is clearer when understanding how this gets rendered by docs:

<img width="731" alt="image" src="https://user-images.githubusercontent.com/34116392/225028372-1a4aa6f4-d1f3-47c1-b57c-6df09ee9f605.png">

Knowing that this gets rendered under "What went wrong?", maybe we could update the description to answer it like:
```suggestion
	 * @description
	 * The endpoint is prerendered with an `undefined` param but the generated file and folder names will collide.
	 * An additional extension can be added to the endpoint file name to generate the file with a different name.
	 * For example, renaming `pages/api/[slug].ts` to `pages/api/[slug].json.ts`.
```

I _think_ this follows the message format. cc @sarah11918 if this is right

Reply:
> Thanks for updating this part @wulinsheng123. I'm checking the errors-data readme again, I think it is clearer when understanding how this gets rendered by docs:
> 
> <img alt="image" width="731" src="https://user-images.githubusercontent.com/34116392/225028372-1a4aa6f4-d1f3-47c1-b57c-6df09ee9f605.png">
> 
> Knowing that this gets rendered under "What went wrong?", maybe we could update the description to answer it like:
> 
> I _think_ this follows the message format. cc @sarah11918 if this is right

My bad.
I will remember it. I'll fix it tomorrow

Reply:
Thanks for your suggestion